### DISCUSSION TITLE
Searches with special characters result in internal server error - mysql db

### DISCUSSION THREAD

**Discussion by njnosko** [wagtail/wagtail#8094] [General]
Search results that have a special character in them are causing internal server errors. The database is a MySQL database. Here’s the error with Debug set to true. The search was for “water & fire” in this instance. Any thoughts on why this is happening?

WAGTAILSEARCH_BACKENDS = {
‘default’: {
‘BACKEND’: ‘wagtail.search.backends.database’,
}
}

and have mysqlclient installed in the python virtual environment

<html><body>
<h1>ProgrammingError at /search/</h1>
<p>(1064, “syntax error, unexpected ‘)’”)</p>
<div class="md-table">

Request Method: | GET
-- | --
Request URL: | https://domainremoved.com/search/?q=water+%26+fire
Django Version: | 3.2.9
Exception Type: | ProgrammingError
Exception Value: | (1064, “syntax error, unexpected ‘)’”)
Exception Location: | /var/www/sites/library-test2/.venv/lib64/python3.8/site-packages/MySQLdb/connections.py, line 259, in query
Python Executable: | /var/www/sites/library-test2/.venv//bin/python
Python Version: | 3.8.8
Python Path: | [’/var/www/sites/library-test2’, ‘/usr/lib64/python38.zip’,  ‘/usr/lib64/python3.8’, ‘/usr/lib64/python3.8/lib-dynload’,  ‘/var/www/sites/library-test2/.venv/lib64/python3.8/site-packages’,  ‘/var/www/sites/library-test2/.venv/lib/python3.8/site-packages’]

</div>
</body>
</html>

**Comment by njnosko** [comment]
After some further tests, only a special character with spaces around it causes the issue. "water& fire" and "water &fire" both work but "water & fire" throws an error.

**Comment by kaedroho** [comment]
Thanks @njn315! Please could you post this as an issue? This looks like it could be a bug

**Comment by berycz** [comment]
so I guess nobody has posted this as an issue, right?
I've just ran into it, when a spam bot sent `?query=💙+Melanie+sent...+💙` (I don't wanna post the whole query in case a github antispam system caught it, or something) - and with that I got `syntax error, unexpected '+'`

**Reply by ACMCMC** [reply]
Hi! Indeed, it looks like a bug. Do other queries that do not use emoji characters (💙) work?

**Reply by berycz** [reply]
I've just tried `ice fire` - works
`ice & fire` / `ice - fire` / `ice + fire` / `ice # fire` - these raise an exception

**Comment by berycz** [comment]
ok, I've posted an issue https://github.com/wagtail/wagtail/issues/8614
